### PR TITLE
[Merged by Bors] - chore(SetTheory/Ordinal/CantorNormalForm): cleanup proofs

### PR DESCRIPTION
**Pull request by vihdzp**
We golf a bunch of proofs, and perform formatting improvements.

We also deprecate `CNF_fst_le`, which is a weaker (and IME no more convenient) form of `CNF_fst_le_log`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Ordinal/CantorNormalForm.lean [hunk 1]
```diff
@@ -76,35 +74,36 @@ theorem CNF_ne_zero {b o : Ordinal} (ho : o ≠ 0) :
...
 /-- Every exponent in the Cantor normal form `CNF b o` is less or equal to `log b o`. -/
 theorem CNF_fst_le_log {b o : Ordinal.{u}} {x : Ordinal × Ordinal} :
     x ∈ CNF b o → x.1 ≤ log b o := by
   refine CNFRec b ?_ (fun o ho H ↦ ?_) o
-  · rw [CNF_zero]
-    intro contra; contradiction
+  · simp
   · rw [CNF_ne_zero ho, mem_cons]
     rintro (rfl | h)
     · exact le_rfl
```
Comment (YnirPaz) [inline Mathlib/SetTheory/Ordinal/CantorNormalForm.lean:102]:
```suggestion
    · rfl
```

File: Mathlib/SetTheory/Ordinal/CantorNormalForm.lean [hunk 2]
```diff
@@ -120,25 +119,27 @@ theorem CNF_lt_snd {b o : Ordinal.{u}} {x : Ordinal × Ordinal} : x ∈ CNF b o
...
-    · rw [h]; simpa only using div_opow_log_lt o hb
+    obtain rfl | h := mem_cons.mp h
+    · exact div_opow_log_lt o hb
     · exact IH h
 
 /-- The exponents of the Cantor normal form are decreasing. -/
 theorem CNF_sorted (b o : Ordinal) : ((CNF b o).map Prod.fst).Sorted (· > ·) := by
   refine CNFRec b ?_ (fun o ho IH ↦ ?_) o
-  · simp only [gt_iff_lt, CNF_zero, map_nil, sorted_nil]
+  · simp
```
Comment (YnirPaz) [inline Mathlib/SetTheory/Ordinal/CantorNormalForm.lean:132]:
```suggestion
  · rw [CNF_zero]
    exact sorted_nil
```

This also works, use the one you prefer.

**Review by YnirPaz — COMMENTED** [review]
A few small things, otherwise good to go.

File: Mathlib/SetTheory/Ordinal/CantorNormalForm.lean (hunk 2 above)
Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/CantorNormalForm.lean:132]: This is short enough that I think it's an improvement.

**Review by YnirPaz — APPROVED** [review]

**Review by b-mehta — APPROVED** [review]
Thanks!

bors merge

**Pull request closed without merging**